File: 1619-path-crossing/path-crossing.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        visitedPositions = set()

        position = (0,0)
        visitedPositions.add(position)

        for direction in path:
            if direction == "N":
                position = (position[0], position[1] + 1)
            elif direction == "S":
                position = (position[0], position[1] - 1)
            elif direction == "E":
                position = (position[0] + 1, position[1])
            else:
                position = (position[0] - 1, position[1])
            
            if position in visitedPositions:
                return True
            visitedPositions.add(position)
        
        return False
        

File: 1619-path-crossing/path-crossing.ts
function isPathCrossing(path: string): boolean {
    
        const visitedPositions: Set<string> = new Set();
        const position = [0,0]
        visitedPositions.add(`${position[0]},${position[1]}`);

        for (const direction of path) {
            switch (direction) {
                case 'N':
                    position[1]++;
                    break;
                case 'S':
                    position[1]--;
                    break;
                case 'E':
                    position[0]++;
                    break;
                case 'W':
                    position[0]--;
                    break;
            }

            const currentPosition: string = `${position[0]},${position[1]}`;
            if (visitedPositions.has(currentPosition)) {
                return true;
            }
            visitedPositions.add(currentPosition);
        }

        return false;
};